IbisSuperBuild/CMakeLists.txt

```diff
@@ -11,19 +11,31 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/../CMakeCommon/DependencyVersions.cmake)
 # Build All External Dependencies except Qt (vtk, Itk, minc, etc.)
 add_subdirectory( ibisExternalDependencies )
 
+OPTION( IBIS_INSTALL_OPENCV "Install OpenCV library needed in some plugins" OFF )
 # Build Ibis
 INCLUDE(ExternalProject)
 set( CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR}/Ibis-install )
 set( VTK_DIR ${CMAKE_CURRENT_BINARY_DIR}/ibisExternalDependencies/vtk-${IBIS_VTK_LONG_VERSION}/install/lib/cmake/vtk-${IBIS_VTK_LONG_VERSION})
 set( ITK_DIR ${CMAKE_CURRENT_BINARY_DIR}/ibisExternalDependencies/itk-${IBIS_ITK_LONG_VERSION}/install/lib/cmake/ITK-${IBIS_VTK_SHORT_VERSION} )
-set( OpenCV_DIR ${CMAKE_CURRENT_BINARY_DIR}/ibisExternalDependencies/opencv-${IBIS_OPENCV_LONG_VERSION}/install/share/OpenCV )
+if( IBIS_INSTALL_OPENCV )
+    set( OpenCV_Installed ON CACHE BOOL "tell CMake that OpenCV is available" )
+    set( OpenCV_DIR ${CMAKE_CURRENT_BINARY_DIR}/ibisExternalDependencies/opencv-${IBIS_OPENCV_LONG_VERSION}/install/share/OpenCV )
+    ExternalProject_Add( Ibis
+                SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../
+                BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/Ibis
+        BUILD_COMMAND   make
+        INSTALL_COMMAND make install
+              CMAKE_ARGS -DCMAKE_OSX_SYSROOT:PATH=${CMAKE_OSX_SYSROOT} -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET} -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX} -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE} -DQt5_DIR:PATH=${Qt5_DIR} -DVTK_DIR:PATH=${VTK_DIR} -DITK_DIR:PATH=${ITK_DIR} -DOpenCV_DIR:PATH=${OpenCV_DIR}
+  )
+else(IBIS_INSTALL_OPENCV )
 ExternalProject_Add( Ibis
 		SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../
 		BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/Ibis
     BUILD_COMMAND   make
     INSTALL_COMMAND make install 
-	  CMAKE_ARGS -DCMAKE_OSX_SYSROOT:PATH=${CMAKE_OSX_SYSROOT} -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET} -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX} -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE} -DQt5_DIR:PATH=${Qt5_DIR} -DVTK_DIR:PATH=${VTK_DIR} -DITK_DIR:PATH=${ITK_DIR} -DOpenCV_DIR:PATH=${OpenCV_DIR}
+          CMAKE_ARGS -DCMAKE_OSX_SYSROOT:PATH=${CMAKE_OSX_SYSROOT} -DCMAKE_OSX_DEPLOYMENT_TARGET:STRING=${CMAKE_OSX_DEPLOYMENT_TARGET} -DCMAKE_INSTALL_PREFIX:PATH=${CMAKE_INSTALL_PREFIX} -DCMAKE_BUILD_TYPE:STRING=${CMAKE_BUILD_TYPE} -DQt5_DIR:PATH=${Qt5_DIR} -DVTK_DIR:PATH=${VTK_DIR} -DITK_DIR:PATH=${ITK_DIR} }
   )
+endif()
```

Comment:
This is quite a lot of code. The only difference between the if and the else is that in one you define OpenCV_DIR and in the other not. I would do something like:

if( IBIS_INSTALL_OPENCV )
    set( OpenCV_DIR ${CMAKE_CURRENT_BINARY_DIR}/ibisExternalDependencies/opencv-${IBIS_OPENCV_LONG_VERSION}/install/share/OpenCV )
    set( IbisOpenCVDefines "-DOpenCV_DIR:PATH=${OpenCV_DIR} -DIBIS_USE_OPENCV:BOOL=TRUE" )
endif()

And then external project add with ${IbisOpenCVDefines} as a CMake_ARGS

I think this would be cleaner. Also, it turns ON IBIS_USE_OPENCV automatically in Ibis if the superbuild builds opencv, which is most likely what users will want.